From 3184d98559e5b9db6bfdc446d1c72ae8c72233a9 Mon Sep 17 00:00:00 2001
From: kholoud mohamed <kholoudm@google.com>
Date: Wed, 12 May 2021 17:06:00 +0100
Subject: [PATCH] Add implementation for UserManager#getAllProfiles

Bug: 183188804
Bug: 183730243
Bug: 195630721
Test: N/A
Change-Id: I7a1893feddf7dadaec30bbffc216b1c41bbd2618
Merged-In: I7a1893feddf7dadaec30bbffc216b1c41bbd2618
(cherry picked from commit 9c7bd44d7e44bbbd1fc664465a4989e23703a1bf)
(cherry picked from commit 35fd431402a9506a0dfc0ff6eb98cfc44cf0d005)
---
 .../main/java/org/robolectric/shadows/ShadowUserManager.java | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/shadows/framework/src/main/java/org/robolectric/shadows/ShadowUserManager.java b/shadows/framework/src/main/java/org/robolectric/shadows/ShadowUserManager.java
index 3223c3ca2..61060f15f 100644
--- a/shadows/framework/src/main/java/org/robolectric/shadows/ShadowUserManager.java
+++ b/shadows/framework/src/main/java/org/robolectric/shadows/ShadowUserManager.java
@@ -119,6 +119,11 @@ public class ShadowUserManager {
     return ImmutableList.copyOf(userProfiles.keySet());
   }
 
+  @Implementation
+  public List<UserHandle> getAllProfiles() {
+    return getUserProfiles();
+  }
+
   /**
    * If any profiles have been added using {@link #addProfile}, return those profiles.
    *
-- 
2.17.1

